src/app/work-item/work-item-list/work-item-detail/work-item-link/work-item-link.component.ts

```diff
@@ -90,98 +85,54 @@ export class WorkItemLinkComponent implements OnInit, OnChanges {
         
     }
 
-    onSelectRelation(relation: LinkType): void{
+    onSelectRelation(relation: any): void{
         this.selectedLinkType = relation;
-        this.searchWorkItem.searchType = relation.relationships.target_type.data.id;
+        // this.searchWorkItem.searchType = relation.relationships.target_type.data.id;
     }
 
     createLink(event: any = null): void {
-         this.createLinkObject(this.workItem['id'], this.selectedWorkItem['id'], this.selectedLinkType['id']);
-         const tempValue = {'data': this.link};
-         this.WorkItemLinkService
-        .createLink(tempValue)
-        .then(response => {
-          this.workitemLinks.push(response);
-          this.calculateTotal();
-        }).catch((error) => {
-            console.log(error);
-        });
+        this.createLinkObject(
+           this.workItem['id'], 
+           this.selectedWorkItem['id'], 
+           this.selectedLinkType.linkId, 
+           this.selectedLinkType.linkType
+          );
+        const tempValue = {'data': this.link};
```

Comment:
The naming is not correct here. It's parameters instead of temporary value. I'll update it.